subprojects/logging/src/main/java/org/gradle/internal/logging/console/StyledTextOutputBackedRenderer.java

```diff
@@ -44,6 +44,9 @@ public void onOutput(OutputEvent event) {
         if (event instanceof LogLevelChangeEvent) {
             LogLevelChangeEvent changeEvent = (LogLevelChangeEvent) event;
             debugOutput = changeEvent.getNewLogLevel() == LogLevel.DEBUG;
+            if (debugOutput && dateFormat == null) {
+                dateFormat = new SimpleDateFormat("HH:mm:ss.SSS");
```

Comment:
:x: Can this be called concurrently? If so, there's a race condition here. A thread can see `dateFormat` not null, while initializing `SimpleDateFormat` is not complete.